src/prognostic_equations/implicit/implicit_solver.jl

```diff
@@ -995,6 +1002,21 @@ function update_implicit_equation_jacobian!(A, Y, p, dtγ)
                     dtγ * ᶠtridiagonal_matrix_c3 ⋅
                     DiagonalMatrixRow(adjoint(CT3(Y.f.sgsʲs.:(1).u₃))) - (I_u₃,)
             end
+
+            # entrainment and detrainment
+            (; ᶜentrʲs, ᶜdetrʲs, ᶜturb_entrʲs) = p
```

Comment:
could you add a comment mentioning the assumption (i.e w/z, dw/dz constant), so it's clear if we change entrainment later